File: R/kentr.R
#' @importFrom Rcpp sourceCpp
#' @importFrom  purrr map_chr map
#' @import dplyr
#' @importFrom stringr str_split str_detect str_replace str_c
#' @import readr
#' @useDynLib kentr, .registration = TRUE
NULL

utils::globalVariables(c(".",
                         "header",
                         "seqnames",
                         "start",
                         "type",
                         "chrom",
                         "end",
                         "transcript_id",
                         "exon_number",
                         "strand"))

File: src/kentr.cpp
#include "kentr.h"

std::unordered_map<char, char> kentr::bpCompTable = {
  {'C', 'G'},
  {'G', 'C'},
  {'T', 'A'},
  {'A', 'T'},
  {'N', 'N'},
  {'n', 'n'},
  {'c', 'g'},
  {'g', 'c'},
  {'t', 'a'},
  {'a', 't'}
};
